File: DesignPatternDemo/Pattern/Response/Women.m
//
//  Women.m
//  DesignPatternDemo
//
//  Created by ruantong on 2019/5/11.
//  Copyright © 2019 wupeng. All rights reserved.
//

#import "Women.h"

@interface Women ()
{
    NSInteger _type;
    NSString * _request;
}
@end

@implementation Women

- (instancetype)initWithType:(LevelType)type request:(NSString *)request{
    self = [super init];
    if (self) {
        _type = type;
        switch (type) {
            case fatherType:
            {
                _request = [NSString stringWithFormat:@"女儿的请求是:%@",request];
            }
                break;
            case husbandType:
            {
                _request = [NSString stringWithFormat:@"妻子的请求是:%@",request];
            }
                break;
            case sonType:
            {
                _request = [NSString stringWithFormat:@"母亲的请求是:%@",request];
            }
                break;
            default:
                break;
        }
        _request = request;
    }
    return self;
}

- (NSInteger)getType{
    return _type;
}

- (NSString *)getRequest{
    return _request;
}

@end

File: DesignPatternDemo/Pattern/Response/Women.swift
//
//  Women.swift
//  DesignPatternDemo
//
//  Created by wupeng on 2020/3/5.
//  Copyright © 2020 wupeng. All rights reserved.
//

import Foundation

enum RequestType:NSInteger{
    case fatherType = 0
    case husbandType = 1
    case sonType = 2
}

protocol IWomen {
    func getType() -> RequestType;
    func getRequest() -> String;
}

class Women: NSObject,IWomen {
    
    var type:RequestType
    var request:String = ""
    
    public init(type:RequestType,request:String) {
        self.type = type
        switch self.type {
        case RequestType.fatherType:
            self.request = "女儿的请求是：\(request)"
            break
        case RequestType.husbandType:
            self.request = "妻子的请求是：\(request)"
            break
        case RequestType.sonType:
            self.request = "母亲的请求是：\(request)"
            break
        default:
            self.request = ""
        }
    }
    
    func getType() -> RequestType {
        return self.type
    }
    
    func getRequest() -> String {
        return self.request
    }
    
}
